provider/core/src/resource.rs

```diff
@@ -34,10 +34,37 @@ impl ResourceKeyHash {
 /// the ones exported by a component.
 #[derive(PartialEq, Eq, Copy, Clone)]
 pub struct ResourceKey {
+    // This string literal is wrapped in leading_tag!() and trailing_tag!() to make it detectable
+    // in a compiled binary.
     path: &'static str,
     hash: ResourceKeyHash,
 }
 
+#[macro_export]
+macro_rules! leading_tag {
```

Comment:
Suggestion, here and below: Make this a const variable instead of a macro.

Reply:
Can't `concat!` const variables.